django/contrib/gis/db/models/aggregates.py

```diff
@@ -8,20 +8,21 @@ class GeoAggregate(Aggregate):
     function = None
     is_extent = False
 
-    def as_sql(self, compiler, connection):
+    def as_sql(self, compiler, connection, function=None, **extra_context):
         # this will be called again in parent, but it's needed now - before
         # we get the spatial_aggregate_name
         connection.ops.check_expression_support(self)
-        self.function = connection.ops.spatial_aggregate_name(self.name)
-        return super().as_sql(compiler, connection)
-
-    def as_oracle(self, compiler, connection):
-        if not hasattr(self, 'tolerance'):
-            self.tolerance = 0.05
-        self.extra['tolerance'] = self.tolerance
-        if not self.is_extent:
-            self.template = '%(function)s(SDOAGGRTYPE(%(expressions)s,%(tolerance)s))'
-        return self.as_sql(compiler, connection)
+        return super().as_sql(
+            compiler,
+            connection,
+            function=function or connection.ops.spatial_aggregate_name(self.name),
+            **extra_context
+        )
+
+    def as_oracle(self, compiler, connection, template=None):
```

Comment:
Should `template=None` be a local variable rather than an (unused) kwarg? I don't think the `as_vendor()` methods typically take any kwargs. Similarly for `function=None` for `as_postgresql/sqlite()`.